src/daemon/rpc_command_executor.cpp

```diff
@@ -1292,6 +1296,40 @@ bool t_rpc_command_executor::print_transaction_pool_stats() {
   return true;
 }
 
+#if defined(SEKRETA)
+bool t_rpc_command_executor::sekreta(
+    const ::sekreta::api::impl_helper::DaemonArgs<std::string>& args)
+{
+  cryptonote::COMMAND_RPC_SEKRETA::request req;
+  cryptonote::COMMAND_RPC_SEKRETA::response res;
+
+  req.command = args.command;
+  req.system = args.system;
+  req.system_args = args.system_args;
+
+  constexpr char fail_message[] = "Sekreta did NOT successfully execute";
+
+  if (m_is_rpc)  // We're getting it from an RPC client, most likely the wallet
+    {
+      if (m_rpc_client->rpc_request(req, res, "/sekreta", fail_message))
+        {
+          tools::success_msg_writer() << "Sekreta executed";
+          return true;
+        }
+      tools::fail_msg_writer() << make_error(fail_message, res.status);
+      return false;
+    }
+
+  if (!m_rpc_server->on_sekreta(req, res) || res.status != CORE_RPC_STATUS_OK)
```

Comment:
That should be an else.

Also, nowhere else do we use GNU indentation, it is quite jarring tbh.